corpora-api.yml

```diff
@@ -0,0 +1,530 @@
+      parameters:
+        - $ref: '#/components/parameters/path_project_uuid'
+      requestBody:
+        content:
+          application/json:
+            schema:
+              description: A schema that has fields for all available project-level metadata.
+              type: object
+              properties:
+                name:
+                  type: string
+                description:
+                  type: string
+                raw_data_link:
+                  type: array
+                  items:
+                    type: string
+                protocol_link:
+                  type: string
+                other_information_link:
+                  type: string
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  project_uuid:
+                    $ref: '#/components/schemas/project_uuid'
+        '400':
+          $ref: '#/components/responses/400'
+        '401':
+          $ref: '#/components/responses/401'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
+
+
+  /v1/submission/{project_uuid}/publish:
+
+    post:
+      tags:
+        - submissions
+      summary: Publish the Project
+      description: >-
+        Publish a project to Corpora as either public or private. The visibility of the project will either make it
+        visible on the public sites or only viewable through obfuscated URLs. If the visibility is set
+        to public, the successful execution of this operation will move the Submission to a CLOSED state.
+      parameters:
+        - $ref: '#/components/parameters/path_project_uuid'
+      requestBody:
+        content:
+          application/json:
+            schema:
+              description: >-
+                a schema that contains a field dictating the visibility of the deployments .
+              type: object
+              properties:
+                visibility:
+                  description: Determines the visibility of the project, that will either make the
+                    project visible on the public sites or only viewable through obfuscated URLs.
+                  type: string
+                  enum: [public,private]
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  project_uuid:
+                    $ref: '#/components/schemas/project_uuid'
+                  visibility:
+                    type: string
+        '400':
+          $ref: '#/components/responses/400'
+        '401':
+          $ref: '#/components/responses/401'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
+
+
+  /v1/projects:
```

Comment:
Noticed the API uses singular for `submission` and `dataset`, but plural for `projects`. Should these be consistent?